bigquery_etl/dryrun.py

```diff
@@ -204,28 +217,30 @@ def dry_run_result(self):
             sql = self.content
         else:
             sql = self.get_sql()
-        if self.metadata:
-            # use metadata to rewrite date-type query params as submission_date
-            date_params = [
-                query_param
-                for query_param in (
-                    self.metadata.scheduling.get("date_partition_parameter"),
-                    *(
-                        param.split(":", 1)[0]
-                        for param in self.metadata.scheduling.get("parameters", [])
-                        if re.fullmatch(r"[^:]+:DATE:{{.*ds.*}}", param)
-                    ),
+
+        query_parameters = []
+        scheduling_metadata = self.metadata.scheduling if self.metadata else {}
+        if date_partition_parameter := scheduling_metadata.get(
+            "date_partition_parameter", "submission_date"
```

Comment:
Does `submission_date` need to be set if if there's no `date_partition_parameter`?

Reply:
If `date_partition_parameter` isn't specified at all in the scheduling metadata then it does default to passing a `@submission_date` parameter, so dryrun should too.  However, if `date_partition_parameter` is set to null in the scheduling metadata then no `@submission_date` parameter is passed by default, so this does the same in that case (the `"date_partition_parameter"` key will exist in the scheduling dictionary, so the `get()` will return the null value as `None` which won't pass the `if` check).

Reply:
Oh I see.  I didn't know it did that